addons/account_qr_code_emv/models/res_bank.py

```diff
@@ -23,25 +25,35 @@ def _get_crc16(self, data, poly=0x1021, init=0xFFFF):
     def _get_merchant_account_info(self):
         return False
 
+    def _get_additional_data_field(self, comment):
+        return False
+
     def _get_qr_vals(self, qr_method, amount, currency, debtor_partner, free_communication, structured_communication):
         if qr_method == 'emv_qr':
             merchant_account_info = self._get_merchant_account_info()
             currency_code = CURRENCY_MAPPING[currency.name]
             merchant_name = self.partner_id.name and self.partner_id.name[:25] or 'NA'
             merchant_city = self.partner_id.city and self.partner_id.city[:15] or ''
+            comment = (free_communication or '') if not structured_communication else ''
```

Comment:
So we don't write the comment if structured?
In the sepa QR code, the structured communication is set separately, unlike here.
In this case, I guess we want to take the structured_communication, something like this:

```suggestion
            comment = structured_communication or free_communication or ''
```